modules/gltf/src/lib/extensions/deprecated/EXT_feature_metadata.ts

```diff
@@ -0,0 +1,142 @@
+/* eslint-disable camelcase */
+import type {GLTF} from '../../types/gltf-types';
+
+import GLTFScenegraph from '../../api/gltf-scenegraph';
+import {
+  ClassProperty,
+  EXR_feature_metadata_class_object,
+  EXT_feature_metadata_feature_table,
+  FeatureTableProperty,
+  GLTF_EXT_feature_metadata
+} from '../../types/gltf-json-schema';
+
+/** Extension name */
+const EXT_FEATURE_METADATA = 'EXT_feature_metadata';
+
+export const name = EXT_FEATURE_METADATA;
+
+export async function decode(gltfData: {json: GLTF}): Promise<void> {
+  const scenegraph = new GLTFScenegraph(gltfData);
+  decodeExtFeatureMetadata(scenegraph);
+}
+
+/** Decode one meshopt buffer view */
+function decodeExtFeatureMetadata(scenegraph: GLTFScenegraph): void {
+  const extension: GLTF_EXT_feature_metadata | null = scenegraph.getExtension(EXT_FEATURE_METADATA);
+  const schemaClasses = extension?.schema?.classes;
+  const featureTables = extension?.featureTables;
+  const featureTextures = extension?.featureTextures;
+
+  if (featureTextures) {
+    /*
+     * TODO add support for featureTextures
+     * Spec - https://github.com/CesiumGS/glTF/tree/3d-tiles-next/extensions/2.0/Vendor/EXT_feature_metadata#feature-textures
+     */
+    // eslint-disable-next-line no-console
+    console.warn('featureTextures is not yet supported in the "EXT_feature_metadata" extension.');
+  }
+
+  if (schemaClasses && featureTables) {
+    for (const schemaName in schemaClasses) {
+      const schemaClass = schemaClasses[schemaName];
+      const featureTable = findFeatureTableByName(featureTables, schemaName);
+
+      if (featureTable) {
+        handleFeatureTableProperties(scenegraph, featureTable, schemaClass);
+      }
+    }
+  }
+}
+
+function handleFeatureTableProperties(
+  scenegraph: GLTFScenegraph,
+  featureTable: EXT_feature_metadata_feature_table,
+  schemaClass: EXR_feature_metadata_class_object
+): void {
+  for (const propertyName in schemaClass.properties) {
+    const schemaProperty = schemaClass.properties[propertyName];
+    const featureTableProperty = featureTable?.properties?.[propertyName];
+    const numberOfFeatures = featureTable.count;
+
+    if (featureTableProperty) {
+      const data = getPropertyDataFromBinarySource(
+        scenegraph,
+        schemaProperty,
+        numberOfFeatures,
+        featureTableProperty
+      );
+      featureTableProperty.data = data;
```

Comment:
The model for these  GLTF extension processors is that they extract data from the extension and then they remove the extension from the tree, as if it had never been there in the first place. 

This works differently and modifies the extension structure and then leaves it in place.

I think an implementation that traverses the extension without modifying it and returns a well typed table would make more sense and be easier to understand. 

Perhaps `readonly scenegraph` parameter to enforce this discipline.

Reply:
Firstly I thought about exacting data from this extension to keep it in another more convenient structure, but as I said before we are having this extension in root file and also in primitives. So follow this logic we should remove it from primitives as well but I don't want to do this because I need this primitives extension data later in `tile-converter`.